schema.production.kf

```diff
@@ -509,6 +734,66 @@ procedure share_credential_by_write_grant(
         );
 }
 
+action share_credential_by_write_grant2(
+  $id
+, $human_id
+, $encryption_public_key
+, $content
+, $public_notes
+, $public_notes_signature
+, $broader_signature
+, $original_credential_id
+, $grantee
+, $locked_until
+, $issuer
+) public {
+    SELECT CASE WHEN $locked_until < 0 THEN
+        error('locked_until must be positive integer timestamp or zero')
+    END;
+
+    SELECT get_write_grant_id($human_id); // throw an error if no write_grant
+
+    SELECT CASE WHEN NOT credential2_belongs_to_human($original_credential_id, $human_id) THEN
+        error('the original credential does not belong to the human')
+    END;
+
+    SELECT CASE WHEN get_write_grant_id($human_id)::text != get_credential2_inserter($original_credential_id) THEN
+        error('you can share only original credentials you created')
+    END;
+
+    SELECT CASE WHEN (SELECT count(1) FROM internal_access_grants WHERE id = uuid_generate_v5('31276fd4-105f-4ff7-9f64-644942c14b79'::uuid, format('%s-%s-%s', $grantee::text, $original_credential_id::text, $locked_until::text))) > 0 THEN
+        error('a grant with the same grantee, original_credential_id, and locked_until already exists')
+    END;
+
+    SELECT CASE WHEN $public_notes != '' THEN
+        error('shared credentials cannot have public_notes, it must be an empty string')
```

Comment:
Oh. From this perspective proposed structure of `credentials2` looks not very good. If we eventually will keep only new credentials, then maybe it is worth to think about split original and shared credentials to different tables?

Reply:
Yeah, that also went through my head. The only reason I didn't do it is that we can always do that change after the fact. For now, I'm foreseeing enough fun on the migration with just the schema changes we already have.